src/layout/Summary2/SummaryComponent2/SummaryComponent2.test.tsx

```diff
@@ -0,0 +1,209 @@
+    await render({
+      summary2Config: {
+        type: 'Summary2',
+        hideEmptyFields: false,
+        id: 'Summary2',
+        target: {
+          id: 'Input',
+          type: 'component',
+        },
+      },
+    });
+    // expect(screen.getByTestId('summary-single-value-component'));
+    expect(screen.getByTestId('summary-single-value-component')).toBeInTheDocument();
+  });
+
+  test('should not render component if its set to hide if empty', async () => {
+    await render({
+      summary2Config: {
+        type: 'Summary2',
+        hideEmptyFields: true,
+        id: 'Summary2',
+        target: {
+          id: 'Input',
+          type: 'component',
+        },
+      },
+    });
+    const element = screen.queryByTestId('summary-single-value-component');
+    expect(element).not.toBeInTheDocument();
+  });
+
+  test('should render component if its set to hide if empty, but the field is required', async () => {
+    await render({
+      layout: {
+        FormLayout: {
+          data: {
+            layout: [
+              {
+                id: 'Input',
+                type: 'Input',
+                dataModelBindings: { simpleBinding: 'field' },
+                required: true,
+              },
+            ],
+          },
+        },
+      },
+      summary2Config: {
+        type: 'Summary2',
+        hideEmptyFields: true,
+        id: 'Summary2',
+        target: {
+          id: 'Input',
+          type: 'component',
+        },
+      },
+    });
+    expect(screen.getByTestId('summary-single-value-component')).toBeInTheDocument();
+  });
+
+  test('should render component if its set to hide if empty, but the component is set to forceShow', async () => {
+    await render({
+      layout: {
+        FormLayout: {
+          data: {
+            layout: [
+              {
+                id: 'Input',
+                type: 'Input',
+                dataModelBindings: { simpleBinding: 'field' },
+                forceShowInSummary: true,
+              },
+            ],
+          },
+        },
+      },
+      summary2Config: {
+        type: 'Summary2',
+        hideEmptyFields: true,
+        id: 'Summary2',
+        target: {
+          id: 'Input',
+          type: 'component',
+        },
+      },
+    });
+    expect(screen.getByTestId('summary-single-value-component')).toBeInTheDocument();
+  });
+
+  test('should hide all empty components if hideEmptyFields=true', async () => {
+    await render({
+      layout: {
+        FormLayout: {
+          data: {
+            layout: [
+              {
+                id: 'Input',
+                type: 'Input',
+                dataModelBindings: { simpleBinding: 'field' },
+                required: true,
+              },
+              {
+                id: 'Input2',
+                type: 'Input',
+                dataModelBindings: { simpleBinding: 'field2' },
+                required: true,
```

Comment:
Would be nice to have one test with `required: false` here as well (which would catch the bug mentioned above).